### PR TITLE
feat(tests) add eks as a target for e2e tests

### PR DESCRIPTION
**Pull request by bartsmykla**
### Summary

Added to our CI EKS as a target for E2E tests

### Documentation

TBD

- [ ] Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -77,6 +83,183 @@ executors:
...
+    # We need kumactl to be built to run tests, and to not build it again,
+    #  we can use the one already build by earlier jobs
+    - attach_workspace:
+        at: build
+
+    - run:
+        name: "Run E2E tests"
+        command: |
+          export KUMA_UNIVERSAL_IMAGE="${ECR_REGISTRY}/kuma-universal:${TAG}";
+          export E2E_PKG_LIST="./test/e2e/helm";
```
Comment (jakubdyszkiewicz) [inline .circleci/config.yml:223]: so for ECS we are running just HELM suite, not full E2E suite?

File: .circleci/config.yml [hunk 2]
```diff
@@ -77,6 +83,183 @@ executors:
...
+          export API_VERSION="v2";
```
Comment (jakubdyszkiewicz) [inline .circleci/config.yml:224]: delete and rely on default (v3)?

File: pkg/envoy/admin/client.go [hunk 3]
```diff
@@ -73,7 +73,7 @@ func (a *envoyAdminClient) GenerateAPIToken(dataplane *mesh_core.DataplaneResour
 }
 
 func (a *envoyAdminClient) getOrCreateSigningKey(mesh string) (string, error) {
-	key, err := issuer.GetSigningKey(a.rm, issuer.DataplaneTokenPrefix, mesh)
+	key, err := issuer.GetSigningKey(a.rm, issuer.EnvoyAdminClientTokenPrefix, mesh)
```
Comment (jakubdyszkiewicz) [inline pkg/envoy/admin/client.go:76]: revert this please. There is a separate PR and it's not that simple change

Comment (bartsmykla) [reply pkg/envoy/admin/client.go:76]: on it

Comment (bartsmykla) [reply pkg/envoy/admin/client.go]: done

File: .circleci/config.yml (hunk 2 above)
Comment (bartsmykla) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 1 above)
Comment (bartsmykla) [reply .circleci/config.yml:223]: Yes, that was requirement I received. To have full suite there would have to be much more changes as we are relying for the "universal" part when doing "hybrid" tests on local machine and it expects some local functionalities which I forgot now

**Review by nickolaev — COMMENTED** [review]
Great job. I believe we are close. Let's find time to chat more.

File: mk/e2e.new.mk [hunk 4]
```diff
@@ -51,19 +55,30 @@ test/e2e/kind/stop: $(K8SCLUSTERS_STOP_TARGETS)
...
+		KUMA_GLOBAL_IMAGE_TAG="$(KUMA_GLOBAL_IMAGE_TAG)" \
+		KUMA_GLOBAL_IMAGE_REGISTRY="$(KUMA_GLOBAL_IMAGE_REGISTRY)" \
+		KUMA_CP_IMAGE_REPOSITORY="$(KUMA_CP_IMAGE_REPOSITORY)" \
+		KUMA_DP_IMAGE_REPOSITORY="$(KUMA_DP_IMAGE_REPOSITORY)" \
+		KUMA_DP_INIT_IMAGE_REPOSITORY="$(KUMA_DP_INIT_IMAGE_REPOSITORY)" \
+		KUMA_USE_LOAD_BALANCER="$(KUMA_USE_LOAD_BALANCER)" \
+		KUMA_IN_EKS="$(KUMA_IN_EKS)" \
+		KUMA_UNIVERSAL_IMAGE="$(KUMA_UNIVERSAL_IMAGE)" \
+		KUMA_WAIT_FOR_MESH_RETRIES="$(KUMA_WAIT_FOR_MESH_RETRIES)" \
+        KUMA_WAIT_FOR_MESH_TIMEOUT="$(KUMA_WAIT_FOR_MESH_TIMEOUT)" \
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply mk/e2e.new.mk:71]: As we discussed on slack, that can be a great future improvements but we are leaving it as it is for now

**Review by nickolaev — APPROVED** [review]

**Pull request merged**